### ISSUE TITLE
Separate adding X-Forwarded-Port header

### ISSUE THREAD

**Issue by gonzalad** [spring-cloud/spring-cloud-netflix#1807]
Similar to #1656 
Add X-Forwarded-Port header in Zuul PreDecorationFilter even with X-Forwarded-Host header being absent.

In my case I deployed my Zuul Proxy gateway behind a Amazon ALBv2.

In my zuul proxy, I receive the following request : 
```
GET /swagger/scim/v2/api-docs
X-Forwarded-For: 10.10.10.10.
X-Forwarded-Proto: https.
X-Forwarded-Port: 443.
Host: front-host.com.
X-Amzn-Trace-Id: Root=blabla
cache-control: max-age=0.
upgrade-insecure-requests: 1.
user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/56.0.2924.76 Chrome/56.0.2924.76 Safari/537.36.
accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8.
accept-encoding: gzip, deflate, sdch, br.
accept-language: en-US,en;q=0.8,fr;q=0.6,es;q=0.4,it;q=0.2.
```

As you can see, there's no X-Forwarded-Host.
In the current code, X-Forwarded-Port is added only if there's a X-Forwarded-Host present.

https://github.com/spring-cloud/spring-cloud-netflix/blob/675e4c1031b275af877938c02d5950c91f7fe660/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/pre/PreDecorationFilter.java#L199

In my case, my Zuul proxy forwards a request containing :
```
x-forwarded-port: 80
```
While I would expect it to send a 
```
x-forwarded-port: 443,80
```

**Comment by michelzanini** [comment]
I have the same problem and was about to open an issue, when I found the bug already reported here.
I also have my service behind an Amazon ELB, which I believe be a very standard use case.

When we do a request to our ELB:

GET https://myelb.domain.com/test

The following headers are sent from the ELB to the Gateway running Zuul:
```
X-Forwarded-For: (the IP address of the ELB)
X-Forwarded-Proto: https 
X-Forwarded-Port: 443
```

Then Zuul forwards the request with:

```
X-Forwarded-Proto: http
X-Forwarded-Host: myelb.domain.com
X-Forwarded-Port: 80
```

And it should have being:

```
X-Forwarded-Proto: https
X-Forwarded-Host: myelb.domain.com
X-Forwarded-Port: 443
```

We added a Apache HTTPD server before Zuul to set X-Forwarded-Host, and that fixes it, as already reported that if X-Forwarded-Host is present, then X-Forwarded-Proto and X-Forwarded-Port are correctly used. However, they should be consider when X-Forwarded-Host is not present as well!

I see there is already an issue open and closed to fix X-Forwarded-Proto and this one is to fix X-Forwarded-Port.
As both are wrong, I would consider this a critical issue.

Looking forward to see this fixed on a service release for Camden, maybe Camden.SR7?

Thanks!

**Comment by ryanjbaxter** [comment]
@gonzalad @michelzanini if either of you want to submit a PR for this, that would be welcome, if not we will get to it when we can

**Comment by gonzalad** [comment]
@ryanjbaxter a PR is available for this issue : #1808

**Comment by michelzanini** [comment]
Thanks @gonzalad. Hopefully this will make into next SR release =]
